File: entregable_profe.py
#Pregunta 1:
#El for es un equivalente al foreach en otros lenguajes que recorre arreglos
#f{} da formato a la cadena de texto a imprimir permitiendoinos embeber una variable
#f es de format y entre las llaves podemos incorporar variables

#Pregunta2:
#recorre un arreglo de cualidades para preguntarte si una persona cumple con esas características, dependiendo de las características que si cumple te da un score
#el swich no existe pero tiene dos equivalentes, elif anidado y match case

#pregunta3:
#en la línea número 3
#hacemos el llamado directo de la función para no hacer más declaraciones de variables que solo se ocuparían una o dos veces en todo el código

#FUNCIONES
def imprimir_cualidades(cualidades):
    print(cualidades)

def agregar_cualidades(cualidades):
    flag=True
    while(flag):
        if(input("Deseas agregar más condiciones?(Y/N)")=="Y"):
            cualidades.append(input("Escribe la propiedad: "))
        else:
            flag=False
        
def cuestionar_cualidades(cualidades):
    score=0
    for cualidad in cualidades:
        prompt=input(f"Es {cualidad}? (Y/N)")
        if(prompt!='N'):
            score=score+1
    return score

def calcular_score(score,len):
    if((score/len)==1):
            print("Es perfecto")
    elif((score/len)>=.8):
            print("Apliquese o se lo ganan")
    elif((score/len)>=.6):
            print("Momeno")
    elif((score/len)>=.4):
            print("No te lo recomiendo")
    elif((score/len)<=.2):
            print("orre y no mires atrás")
#VARIABLES
qualities=["alto","bronceado","guapo","rico","sexy","tierno","rudo"]

#programa

print("Vamos a ver si la persona que piensas te conviene")
print("Estas son las cualidades que actualmente validamos")
imprimir_cualidades(qualities)
agregar_cualidades(qualities)
print("vamos a comenzar")

score = cuestionar_cualidades(qualities)
calcular_score(score,len(qualities))

## comandos de git para agregar el archivo

# 1 git add entregable_profe.py
# 2 git commit -m "entregable"
# 3 git push origin main

File: entregable_profe.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {

    //FUNCIONES
    static void imprimirCualidades(List<String> cualidades) {
        for (String cualidad : cualidades) {
            System.out.println(cualidad);
        }
    }

    static void agregarCualidades(List<String> cualidades) {
        boolean flag = true;
        Scanner scanner = new Scanner(System.in);
        while (flag) {
            System.out.println("Deseas agregar más condiciones?(Y/N)");
            if (scanner.nextLine().equals("Y")) {
                System.out.println("Escribe la propiedad: ");
                cualidades.add(scanner.nextLine());
            } else {
                flag = false;
            }
        }
    }

    static int cuestionarCualidades(List<String> cualidades) {
        int score = 0;
        Scanner scanner = new Scanner(System.in);
        for (String cualidad : cualidades) {
            System.out.println("Es " + cualidad + "? (Y/N)");
            if (!scanner.nextLine().equals("N")) {
                score++;
            }
        }
        return score;
    }

    static void calcularScore(int score, int len) {
        double ratio = (double) score / len;
        if (ratio == 1) {
            System.out.println("Es perfecto");
        } else if (ratio >= 0.8) {
            System.out.println("Apliquese o se lo ganan");
        } else if (ratio >= 0.6) {
            System.out.println("Momeno");
        } else if (ratio >= 0.4) {
            System.out.println("No te lo recomiendo");
        } else if (ratio <= 0.2) {
            System.out.println("Corre y no mires atrás");
        }
    }

    public static void main(String[] args) {
        //VARIABLES
        List<String> qualities = new ArrayList<>();
        qualities.add("alto");
        qualities.add("bronceado");
        qualities.add("guapo");
        qualities.add("rico");
        qualities.add("sexy");
        qualities.add("tierno");
        qualities.add("rudo");

        //PROGRAMA
        System.out.println("Vamos a ver si la persona que piensas te conviene");
        System.out.println("Estas son las cualidades que actualmente validamos");
        imprimirCualidades(qualities);
        agregarCualidades(qualities);
        System.out.println("Vamos a comenzar");

        int score = cuestionarCualidades(qualities);
        calcularScore(score, qualities.size());
    }
}
